heksher/db_logic/util.py

```diff
@@ -0,0 +1,60 @@
+from typing import NamedTuple, Collection, Tuple, TypeVar, Generic, Sequence, Hashable, Dict, Union, Literal
+
+T = TypeVar('T', bound=Hashable)
+
+
+# pytype: disable=not-supported-yet
+
+class SupersequenceResults(NamedTuple, Generic[T]):
+    """
+    An object returned by is_supersequence to denote that the supersequence matches, along with the elements in the
+    supersequence that are missing from the subsequence
+    """
+    new_elements: Collection[Tuple[T, int]]
+
+
+def is_supersequence(supersequence: Sequence[T], subsequence: Collection[T]) \
```

Comment:
```suggestion
def is_supersequence(supersequence: Sequence[T], subsequence: Sequence[T]) \
```

Reply:
Actually no, since we pass a dict to this function (we use the values elsewhere). We depend on the subsequence being ordered, but we don't need it to be a sequence.

Reply:
Sequence is in the name. Besides, the fact dict keeps order is an anomaly, right now you tell anyone using this function that they can put in a set. Either change to sequence and give it only the keys in a tuple, or document that it must have an order (which is probably what you'll do, so I hate you for just think about it while reading this wall of text)

Reply:
Sequence is in the name since that is the mathematical name for this relation, it has nothing to do with python types.

Reply:
Who said anything about Python types? It's in the name because it's a sequence. Otherwise it'd be a superset. (Which by the way, has nothing to do with the Python type `set`. They just "happen" to be related somehow 😋 ).